crates/pop-cli/src/commands/build/spec.rs

```diff
@@ -0,0 +1,423 @@
+	pub(crate) genesis_state: bool,
+	/// Whether the genesis code file should be generated [default: true].
+	#[clap(long = "genesis-code", default_value = "true")]
+	pub(crate) genesis_code: bool,
+}
+
+impl BuildSpecCommand {
+	/// Executes the command.
+	pub(crate) async fn execute(self) -> anyhow::Result<&'static str> {
+		// Checks for appchain project in `./`.
+		if is_supported(None)? {
+			// If para id has been provided we can build the spec
+			// otherwise, we need to guide the user.
+			let _ = match self.id {
+				Some(_) => self.build(&mut cli::Cli),
+				None => {
+					let config = guide_user_to_generate_spec().await?;
+					config.build(&mut cli::Cli)
+				},
+			};
+			return Ok("spec");
+		} else {
+			cli::Cli.outro_cancel(
+				"🚫 Can't build a specification for target. Maybe not a chain project ?",
+			)?;
+			Ok("spec")
+		}
+	}
+
+	/// Builds a parachain spec.
+	///
+	/// # Arguments
+	/// * `cli` - The CLI implementation to be used.
+	fn build(self, cli: &mut impl cli::traits::Cli) -> anyhow::Result<&'static str> {
+		cli.intro(format!("Building your chain spec"))?;
+
+		// Either a para id was already provided or user has been guided to provide one.
+		let para_id = self.id.unwrap_or(DEFAUTL_PARA_ID_VALUE);
+		// Notify user in case we need to build the parachain project.
+		if !self.release {
+			cli.warning("NOTE: this command defaults to DEBUG builds for development chain types. Please use `--release` (or simply `-r` for a release build...)")?;
+			#[cfg(not(test))]
+			sleep(Duration::from_secs(3))
+		}
+
+		let spinner = cliclack::spinner();
+		spinner.start("Generating chain specification...");
+
+		// Create output path if needed
+		let mut output_path = self.output_file.unwrap_or_else(|| PathBuf::from("./"));
+		let mut plain_chain_spec;
+		if output_path.is_dir() {
+			if !output_path.exists() {
+				// Generate the output path if needed
+				create_dir_all(&output_path)?;
+			}
+			plain_chain_spec = output_path.join(DEFAULT_SPEC_NAME);
+		} else {
+			plain_chain_spec = output_path.clone();
+			output_path.pop();
+			if !output_path.exists() {
+				// Generate the output path if needed
+				create_dir_all(&output_path)?;
+			}
+		}
+		plain_chain_spec.set_extension("json");
+
+		// Locate binary, if it doesn't exist trigger build.
+		let mode: Profile = self.release.into();
+		let cwd = current_dir().unwrap_or(PathBuf::from("./"));
+		let binary_path = match binary_path(&mode.target_folder(&cwd), &cwd.join("node")) {
+			Ok(binary_path) => binary_path,
+			_ => {
+				cli.info(format!("Node was not found. The project will be built locally."))?;
+				cli.warning("NOTE: this may take some time...")?;
+				build_parachain(&cwd, None, &mode, None)?
+			},
+		};
+
+		// Generate plain spec.
+		spinner.set_message("Generating plain chain specification...");
+		let mut generated_files =
+			vec![format!("Specification and artifacts generated at: {}", &output_path.display())];
```

Comment:
Sorry for another change. But do you think makes sense to remove this message? `format!("Specification and artifacts generated at: {}", &output_path.display())`

I tried `pop build spec` in the parachain, and I see:
```
◆  Generated files:
│  ● Specification and artifacts generated at: .
```
Which seems a bit weird

Reply:
Yeah fair